### DISCUSSION TITLE
Is there a Bevy equivalent to JMonkeyEngine's SkinningControl class?

### DISCUSSION THREAD

**Discussion by JustBobinAround** [bevyengine/bevy#7338] [Q&A] [answered]
Is there a Bevy equivalent to JMonkeyEngine's SkinningControl class? This class allowed for the direct manipulation of armature joints. For instance:

 `Spatial someImportedGLTF = new Spatial();`
`SkinningControl sc = someImportedGLTF.getControl(SkinningControl.class);`
`sc.getAttachmentsNode("Left_hand").rotate(90, 0, 0);`

In this case `someImportedGLTF` would be a character, and we would be rotating the `"Left_hand"` of the armature which changes the character's mesh as well. I understand that bevy has skeletal animations using `AnimationPlayer`. However, from what I've seen, this only plays prior made animations stored in a animations struct containing a `Handle<AnimationClip>`.

Essentially, is there any way to access an imported glTF object's armature for direct manipulation instead of playing animations?

**Comment by tim-blackbird** [comment, marked as answer]
Yes. Once a gltf scene is spawned the armature will exist in the world as a hierarchy of entities(bones).
You can create a system that looks up the entity by name("Left_hand" in this case) by querying for the `Name` component and editing its `Transform`.
The AnimationPlayer itself is actually implemented in this way [link](https://github.com/bevyengine/bevy/blob/87722d135fcc7e6058aa4710e7fe5be5f2a6e594/crates/bevy_animation/src/lib.rs#L186).

**Reply by JustBobinAround** [reply]
Very big help. Thank you!!
